### PR TITLE
annotations updates for Openlayers

### LINKED ISSUE #2529: Update Annotation layout
### Description
Align the annotation's layout/style with [mockup](https://demo.geo-solutions.it/mockups/mapstore2/#/annotations)'s one

### Acceptance Criteria
The new layout/style must be applied.

### LINKED ISSUE #2531: Add lines and polygons for Annotations
### Description
Add the possibility to add lines and polygons to a single annotation (Multi Geometry)

### AC
- You can draw multiple lines at the same time
- You can draw multiple polygons at the same time
- You can switch from the geometry types and draw with the selected type

### LINKED ISSUE #2532: Add Text for Annotations
### Description
Allow to add text for the Annotation

### PR DESCRIPTION
**Pull request by MV88**
## Description
Added possibility to create, edit, delete Annotations containing:
- Lines
- Polygons
- Text (points)
- Marker (points)

## Issues
 - Fix #2529
 - Fix #2531
 - Fix #2532

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
Actually you can only add, edit remove marker for Annotations

**What is the new behavior?**
This PR add the possibility to draw and edit feature of type **GeometryCollection** containing (markers, lines, polygons and text)

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: 

**Other information**:
There is a retrocompatibility for old style structure, but if in some cases an error occurs is probably related to this change.

old style structure :
```
{
  iconGlyph: "comment",
  ...
}
```
new style structure
```
{
  "MultiLineString": {}
  "Point" : {
    iconGlyph: "comment",
    ...
  },
  type: "GeometryCollection"
}
or
{
  "MultiLineString": {},
  type: "MultiLineString"
}
```

**Note:** for future development use the new style structure

### CODE REVIEW

File: web/client/components/map/openlayers/DrawSupport.jsx [hunk 1]
```diff
@@ -105,20 +109,26 @@ class DrawSupport extends React.Component {
                 if (f.style) {
                     let olFeature = this.toOlFeature(f);
                     if (olFeature) {
-                        olFeature.setStyle(this.toOlStyle(f.style, f.selected));
+                        olFeature.setStyle(f.style && f.style.type ? VectorStyle.getStyle(f) : this.toOlStyle(f.style, f.selected));
```
Comment (mbarto) [inline web/client/components/map/openlayers/DrawSupport.jsx]: For the future: we should try to remove toOlStyle if possible and always use VectorStyle.getStyle

File: web/client/components/map/openlayers/DrawSupport.jsx [hunk 2]
```diff
@@ -105,20 +109,26 @@ class DrawSupport extends React.Component {
...
             });
         }
     };
 
     addLayer = (newProps, addInteraction) => {
+        let layerStyle = null;
+        if (newProps.style) {
+            layerStyle = newProps.style.type ? VectorStyle.getStyle(newProps, false, newProps.features[0] && newProps.features[0].properties && newProps.features[0].properties.textValues || [] ) : this.toOlStyle(newProps.style, null, newProps.features[0] && newProps.features[0].type);
+        } else {
+            layerStyle = VectorStyle.getStyle({...newProps, style: {...DEFAULT_ANNOTATIONS_STYLES, type: this.convertGeometryTypeToStyleType(newProps.drawMethod) }}, false, newProps.features[0] && newProps.features[0].properties && newProps.features[0].properties.textValues || [] );
```
Comment (mbarto) [inline web/client/components/map/openlayers/DrawSupport.jsx]: DrawSupport should not be aware of annotations (DEFAULT_ANNOTATIONS_STYLES)

File: web/client/components/map/openlayers/VectorStyle.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 var markerIcon = require('./img/marker-icon.png');
 var markerShadow = require('./img/marker-shadow.png');
+const {DEFAULT_ANNOTATIONS_STYLES} = require('../../../utils/AnnotationsUtils');
```
Comment (mbarto) [inline web/client/components/map/openlayers/VectorStyle.js]: annotations stuff does not belong to this

**Comment by MV88** [comment]
For the tests 
(start from new map)
create a new annotation (any geom) : 
-  and save it
- and change style save it
- and change style do not save it ( a modal should appear ) click confirm to discard changes
- and change style do not save it ( a modal should appear ) click cancel to keep changes then save annotation

edit one existing annontation with : 
- only markers
- only lines
- every geometry combination

try to delete geometries and add new ones (test all kind of geom after deletion)

try to delete an annotation

try to create an annotation , save the map as new, enter the map edit the annotation save the map (update it, not new) reload and verify changes are present

test that the draw tool is a toggle button between drawing functionality and editing of the annotation
when toggled (green) you should be able to select a type of geom when not toggled (not greeen) you should be able to edit the annotation geometries and forms

all the previous use of the drawing functionality in the query for and feature grid should work as expected with no changes from the previous implementation


verify that opening the feature grid the annotations panels closes and vice versa
i just noticed that if you have feature grid opened and you open the annotations the feature grid stays open

**Pull request merged**